tools/test.py

```diff
@@ -211,10 +212,20 @@ def RunSingle(self, parallel, thread_id):
       if output.UnexpectedOutput():
         if FLAKY in output.test.outcomes and self.flaky_tests_mode == DONTCARE:
           self.flaky_failed.append(output)
+        elif FLAKY in output.test.outcomes and self.flaky_tests_mode == KEEP_RETRYING:
+          for _ in range(99):
+            if not case.Run().UnexpectedOutput():
+              self.flaky_failed.append(output)
+              break
+          # If after 100 tries, the test is not passing, it's not flaky.
+          self.failed.append(output)
```

Comment:
I don't understand this. Doesn't this mark all flaky tests as failed, regardless of if they pass one of the tries? does this need a [for-else](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops)?

Reply:
It does, thanks, good catch :)